File: user-app/src/main/java/com/github/dkurata38/liblio/user_app/security/LiblioUser.java
package com.github.dkurata38.liblio.user_app.security;

import lombok.Getter;

import java.util.Collection;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

public class LiblioUser extends User {
	@Getter
	private final String userId;

	public LiblioUser(String userId, String username, String password, Collection<? extends GrantedAuthority> authorities) {
		super(username, password, authorities);
		this.userId = userId;
	}

	public LiblioUser(String userId, String username, String password, boolean enabled, boolean accountNonExpired, boolean credentialsNonExpired, boolean accountNonLocked, Collection<? extends GrantedAuthority> authorities) {
		super(username, password, enabled, accountNonExpired, credentialsNonExpired, accountNonLocked, authorities);
		this.userId = userId;
	}
}

File: user-app/src/main/kotlin/com/github/dkurata38/liblio/user_app/security/LiblioUser.kt
package com.github.dkurata38.liblio.user_app.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User

class LiblioUser(
    val userId: String,
    username: String,
    password: String,
    enabled: Boolean,
    accountNonExpired: Boolean,
    credentialsNonExpired: Boolean,
    accountNonLocked: Boolean,
    authorities: MutableCollection<out GrantedAuthority>
) : User(username, password, enabled, accountNonExpired, credentialsNonExpired, accountNonLocked, authorities) {
    constructor(userId: String, username: String,
                password: String, authorities: MutableCollection<out GrantedAuthority>): this(userId, username, password, true, true, true, true, authorities) {
    }
}
